torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1301,6 +1301,50 @@ struct to_ir {
     return list_value;
   }
 
+  Value* emitDictComprehension(const DictComp& dc, const TypePtr& type_hint) {
+    const auto loc = dc.range();
+    const auto targets_list = List<Expr>::create(dc.range(), {dc.target()});
+    const auto itrs = List<Expr>::create(dc.range(), {dc.iter()});
+
+    Value* dict_value =
+        graph->insertNode(graph->create(prim::DictConstruct, 1))->output();
```

Comment:
The "unrolled loop" thing is for heterogenous types which we want to allow to be iterated over. 
For example: 
```
for elem in (1, 1.5, "abc"):
    print(elem)
```
It also applies with ModuleLists, which contain different Modules, which are different types.
```[module(x) for module in self.module_list]```. You could have an empty tuple  or an empty modulelist, so what would the output type be?  `[elem for elem in tuple()]`. We could either  throw or give the same default as an  empty list constructor. For lists, we set it as the type of empty list constructor `x = []`. I think we should do  the same in dict comprehensions. 

All that entails is setting the DictConstruct initial type here to be [Str, Tensor] (same `x = {}`).